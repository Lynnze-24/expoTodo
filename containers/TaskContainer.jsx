import React from 'react'
import { View,StyleSheet } from 'react-native'
import cusTheme from '../theme/Theme'

const TaskContainer = ({children,testID}) => {
    const taskid = testID? testID:'TaskCon'; 
    return (
        <View testID={taskid} style={styles.container}>
            {children}
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
      backgroundColor:cusTheme.bg,
      flex:1,
      paddingHorizontal:20
    },
})

export default TaskContainer;
