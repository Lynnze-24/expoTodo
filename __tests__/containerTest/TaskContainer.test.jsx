import { render, fireEvent } from '@testing-library/react-native';
import { Text } from 'react-native-paper';
import TaskContainer from '../../containers/TaskContainer';
import cusTheme from '../../theme/Theme';
import renderer from 'react-test-renderer';
it('renders correctly', () => {
    const comp = renderer.create((<TaskContainer></TaskContainer>)).toJSON();
    expect(comp).toMatchSnapshot();
});

it('should wrap child components correctly',()=>{
    
    const { getByText,getByTestId}  = render(
           <TaskContainer>
               <Text>Hello</Text>
           </TaskContainer>
           
            );
            getByText('Hello') // testing if child is inside
            const taskCon =   getByTestId('TaskCon')
           expect(taskCon).toHaveStyle({
            backgroundColor:cusTheme.bg,
            flex:1,
            paddingHorizontal:20
          }); // testing if con have correct styles

})
