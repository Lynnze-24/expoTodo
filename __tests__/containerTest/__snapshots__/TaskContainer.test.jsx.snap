// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fffeee",
      "flex": 1,
      "paddingHorizontal": 20,
    }
  }
  testID="TaskCon"
/>
`;
